includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1357,17 +1357,18 @@ protected function build_display_items( $itemized_display_items = false ) {
 		$items_total = wc_format_decimal( WC()->cart->cart_contents_total, WC()->cart->dp ) + $discounts;
 		$order_total = version_compare( WC_VERSION, '3.2', '<' ) ? wc_format_decimal( $items_total + $tax + $shipping - $discounts, WC()->cart->dp ) : WC()->cart->get_total( '' );
 
-		if ( wc_tax_enabled() ) {
+		if ( wc_tax_enabled() && 'incl' !== get_option( 'woocommerce_tax_display_cart' ) ) {
```

Comment:
You might need to check whether `woocommerce_tax_display_shop` is inclusive of tax for handling express payments from product page.

Reply:
Resolved inconsistencies by using `woocommerce_tax_display_checkout` to control product price as well as tax line on the pop-up. Using different variables may lead to inconsistent calculations within the pop-up for merchants who set differing values between `woocommerce_tax_display_shop` and `woocommerce_tax_display_checkout`, even though that is not recommended.